src/Ztm.Zcoin.Synchronization/AddressWatcher.cs

```diff
@@ -158,9 +158,8 @@ public class AddressWatcher : ITransactionConfirmationListener
                 }
 
                 // Add debit to address.
-                balances[address] = balances.ContainsKey(address)
-                    ? balances[address] + spend.Value
-                    : spend.Value;
+                Money val;
+                balances[address] = balances.TryGetValue(address, out val) ? val + spend.Value : spend.Value;
```

Comment:
IMO better to rename `val` to `previous` will be better.

Reply:
make sense